libraries/anvill_passes/src/RecoverStackFrameInformation.cpp

```diff
@@ -274,18 +300,51 @@ RecoverStackFrameInformation::UpdateFunction(
   // Pre-initialize the stack frame to zero if we have been requested
   // to do so. From the whole FunctionPass class, we get the setting
   // from the LiftingOptions class
-  if (init_strategy == StackFrameStructureInitializationProcedure::kZeroes) {
-    auto null_value = llvm::Constant::getNullValue(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+  switch (init_strategy) {
+    case StackFrameStructureInitializationProcedure::kZeroes: {
+      auto null_value = llvm::Constant::getNullValue(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
 
-  } else if (init_strategy ==
-             StackFrameStructureInitializationProcedure::kUndef) {
-    auto null_value = llvm::UndefValue::get(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+      break;
+    }
 
-  } else if (init_strategy !=
-             StackFrameStructureInitializationProcedure::kNone) {
-    return StackAnalysisErrorCode::InvalidParameter;
+    case StackFrameStructureInitializationProcedure::kUndef: {
+      auto null_value = llvm::UndefValue::get(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
+
+      break;
+    }
+
+    case StackFrameStructureInitializationProcedure::kSymbolic: {
+      auto current_offset = stack_frame_analysis.lowest_offset;
+      auto &module = *function.getParent();
+      auto byte_type = llvm::Type::getInt8Ty(module.getContext());
+
+      for (auto i = 0U; i < stack_frame_analysis.size; ++i) {
+        auto stack_frame_byte = builder.CreateGEP(
+            stack_frame_alloca,
+            {builder.getInt32(0), builder.getInt32(0), builder.getInt32(i)});
+
+        auto symbolic_value_res =
+            GetStackSymbolicByteValue(module, current_offset);
+
+        if (!symbolic_value_res.Succeeded()) {
+          return symbolic_value_res.TakeError();
+        }
+
+        auto symbolic_value = symbolic_value_res.TakeValue();
+        ++current_offset;
+
+        auto casted_sym_value =
+            llvm::ConstantExpr::getPtrToInt(symbolic_value, byte_type);
+
+        builder.CreateStore(casted_sym_value, stack_frame_byte);
+      }
+
+      break;
+    }
+
+    case StackFrameStructureInitializationProcedure::kNone: break;
   }
 
   // The stack analysis we have performed earlier contains all the
```

Comment:
Can you add something after this along the lines of: if we're in a debug build, then run a new analysis, and it should fail to find to find any stack-related operations. This might help us detect unusual corner cases in the future.

Reply:
There is a test that covers this:

```c++
          // If we run a second stack analysis, we should no longer find any
          // stack frame operation to recover
          stack_frame_analysis_res = RecoverStackFrameInformation::AnalyzeStackFrame(function);
          REQUIRE(stack_frame_analysis_res.Succeeded());

          auto second_stack_frame_analysis = stack_frame_analysis_res.TakeValue();
          CHECK(second_stack_frame_analysis.stack_operation_list.empty());
```